pkg/apiserver/api_installer.go

```diff
@@ -689,6 +709,9 @@ var _ ScopeNamer = legacyScopeNaming{}
 
 // Namespace returns the namespace from the query or the default.
 func (n legacyScopeNaming) Namespace(req *restful.Request) (namespace string, err error) {
+	if n.scope.Name() == meta.RESTScopeNameRoot {
```

Comment:
Does this break events, which can either have a namespace or not depending on what they are for?